services/galley/src/Galley/API/LegalHold.hs

```diff
@@ -333,27 +342,88 @@ disableForUser zusr tid uid (Public.DisableLegalHoldForUserRequest mPassword) =
       . Log.field "action" (Log.val "LegalHold.disableForUser")
   zusrMembership <- Data.teamMember tid zusr
   void $ permissionCheck ChangeLegalHoldUserSettings zusrMembership
-  uidMembership <- Data.teamMember tid uid
-  if not $ userLHEnabled uidMembership
+
+  userLHStatus <- maybe defUserLegalHoldStatus (view legalHoldStatus) <$> Data.teamMember tid uid
+  if not $ userLHEnabled userLHStatus
     then pure DisableLegalHoldWasNotEnabled
-    else disableLH $> DisableLegalHoldSuccess
+    else disableLH userLHStatus $> DisableLegalHoldSuccess
   where
     -- If not enabled nor pending, then it's disabled
-    userLHEnabled :: Maybe TeamMember -> Bool
-    userLHEnabled target = do
-      case fmap (view legalHoldStatus) target of
-        Just UserLegalHoldEnabled -> True
-        Just UserLegalHoldPending -> True
-        Just UserLegalHoldDisabled -> False
-        Just UserLegalHoldNoConsent -> False
-        Nothing -> {- user not found -} False
-
-    disableLH :: Galley ()
-    disableLH = do
+    userLHEnabled :: UserLegalHoldStatus -> Bool
+    userLHEnabled = \case
+      UserLegalHoldEnabled -> True
+      UserLegalHoldPending -> True
+      UserLegalHoldDisabled -> False
+      UserLegalHoldNoConsent -> False
+
+    disableLH :: UserLegalHoldStatus -> Galley ()
+    disableLH userLHStatus = do
       ensureReAuthorised zusr mPassword
       Client.removeLegalHoldClientFromUser uid
       LHService.removeLegalHold tid uid
       -- TODO: send event at this point (see also: related TODO in this module in
       -- 'approveDevice' and
       -- https://github.com/wireapp/wire-server/pull/802#pullrequestreview-262280386)
-      LegalHoldData.setUserLegalHoldStatus tid uid UserLegalHoldDisabled
+      changeLegalholdStatus tid uid userLHStatus UserLegalHoldDisabled
+
+changeLegalholdStatus :: TeamId -> UserId -> UserLegalHoldStatus -> UserLegalHoldStatus -> Galley ()
+changeLegalholdStatus tid uid oldLhStatus lhStatus = do
+  case (hasLH oldLhStatus, hasLH lhStatus) of
+    (False, False) -> pure ()
+    (True, True) -> pure ()
+    (False, True) -> do
+      blockConnectionsFrom1on1s
+      LegalHoldData.setUserLegalHoldStatus tid uid lhStatus
+    (True, False) -> do
+      LegalHoldData.setUserLegalHoldStatus tid uid lhStatus
+      void $ putConnectionInternal (RemoveLHBlocksInvolving uid)
+  where
+    hasLH :: UserLegalHoldStatus -> Bool
```

Comment:
code duplication: `hasLhStatus` here and `userLHEnabled` from line 352 in the same file have the same definition.

Reply:
fixed: da45b70af1feee8317d6ff314bed693e945e54a8